app/models/ems_folder.rb

```diff
@@ -160,10 +160,24 @@ def register_host(host)
   #     :exclude_non_display_folders => false
   def folder_path_objs(*args)
     options = args.extract_options!
-    folders = path(:of_type => "EmsFolder")
-    folders = folders[1..-1] if options[:exclude_root_folder]
-    folders = folders.reject(&:hidden?) if options[:exclude_non_display_folders]
-    folders
+    self.class.folder_options path(:of_type => "EmsFolder"), options
```

Comment:
I'm surprised the bot is not picking up on the lack of parens on the method call (i.e. please use parens).